src/pages/CountryDetails/CountryDetails.tsx

```diff
@@ -0,0 +1,110 @@
+import { useParams, Link, useNavigate } from 'react-router-dom';
+import { useFetch } from '../../pages/Countries/hooks/useFetch';
+import { CountryDetails } from '@types/types';
+import { Toast } from '@/components/Toast/Toast';
+import { LoadingOverlay } from '@/components/LoadingOverlay/LoadingOverlay';
+
+import './CountryDetails.scss';
+
+export function CountryDetail() {
+  const { countryCode } = useParams();
+  const navigate = useNavigate();
+
+  const {
+    data: country,
+    loading,
+    error,
+  } = useFetch<CountryDetails>(
+    `https://restcountries.com/v3.1/alpha/${countryCode}?fields=name,population,region,subregion,capital,flags,cca3,borders,currencies,languages,tld`
```

Comment:
zdaje się, że część tego urla może być taka sama jak w przypadku listy krajów - możesz to wrzucić w osobny plik (zwyczajowo 'api.ts')